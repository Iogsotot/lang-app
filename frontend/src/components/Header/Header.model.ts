export interface HeaderProps {
  smt?: any,
}
