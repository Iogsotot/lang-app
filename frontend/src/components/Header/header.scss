@import "../../variables";

header {
  border-color: white;
  box-shadow: 0 0.25rem 0.625rem -0.125rem rgba(0, 0, 0, 0.1);
  height: 85.2px;

  &__logo-field {
    display: none;
  }
}

.header__inner {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1.1rem 1.1rem 0.8rem;
  height: 85.2px;
  .header__logo-field {
    display: none;
  }
}

.nav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  &__link {
    color: var(--bg-text-subtitle);
    margin: 0 1rem;
    transition: all 0.2s linear;
    border-bottom: 0.125rem solid transparent;
    &:hover {
      color: $color_important;
      transition: all 0.2s linear;
      text-shadow: 0 0 0 rgba(0, 0, 0, 0.5);
      border-bottom: 0.125rem solid $color_important;
    }
  }
}
.dictionary__btn {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 46px;
  transition: all 0.3s linear;
  &:hover {
    filter: brightness(0.5);
    transition: all 0.3s linear;
  }
  &.active {
    filter: hue-rotate(75deg);
    &:hover {
      filter: hue-rotate(75deg) brightness(0.5);
      transition: all 0.3s linear;
    }
  }
  img {
    display: block;
    max-width: 100%;
  }
}
@media (max-width: 640px) {
  .header__inner {
    flex-wrap: wrap;
    padding: 0.5rem;
    .header__logo-field {
      display: inline;
      order: 1;
      img {
        height: 4rem;
        margin-left: 0.9rem;
      }
    }
    .nav {
      order: 3;
    }
    .user__menu {
      order: 2;
      .btn {
        font-size: 16px;
        padding: 10px 40px;
        margin-bottom: 0.5rem;
      }
    }
  }
}

@media (max-width: 930px) {
  header {
    height: 100px;
  }
  .header__inner {
    flex-wrap: wrap;
  }
}
