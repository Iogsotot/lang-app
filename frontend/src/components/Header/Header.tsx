import './header.scss';
import { FC, useEffect, useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import { useTypedSelector } from '../../hooks/useTypedSelector';
import { useAction } from '../../hooks/useAction';
import avatarHolder from '../../assets/icons/avatar-holder.png';
import logoutIcon from '../../assets/icons/logout.png';
import dictionaryIcon from '../../assets/images/dictionary_icon.png';
import { LOCATIONS } from '../../constants';

const { dictionary } = LOCATIONS;

const Header: FC = () => {
  const history = useHistory();
  const { isLoggedIn, user } = useTypedSelector(store => store.user);
  const { avatar, email, name } = user;
  const { logout } = useAction();
  const [isDictionaryPage, setIsDictionaryPage] = useState(false);

  useEffect(() => history.listen(() => {
    if (history?.location?.pathname?.split('/')[1] === dictionary) {
      setIsDictionaryPage(true);
    } else {
      setIsDictionaryPage(false);
    }
  }), [history]);

  return (
    <header>
      <div className="header__inner">
        <nav className="nav">
          <a href="/#features" className="nav__link">
            Возможности
          </a>
          <a href="/#video" className="nav__link">
            Видеообзор
          </a>
          <a href="/#team" className="nav__link">
            Команда
          </a>
          <a href="/#games" className="nav__link">
            Игры
          </a>
        </nav>

        <div className="user__menu">
          {isLoggedIn ? (
            <>
              <Link
                className={`dictionary__btn ${isDictionaryPage ? 'active' : ''}`}
                to="/dictionary/learning/0/0"
              >
                <img src={dictionaryIcon} alt=""/>
              </Link>
              <div className="user">
                <div className="user__info">
                  <p>{name}</p>
                  <p>{email}</p>
                </div>
                <div className="user__icon">
                  <img src={avatar || avatarHolder} alt="" />
                </div>
              </div>
              <button className="user__logout" onClick={logout}>
                <img src={logoutIcon} alt="" />
              </button>
            </>
          ) : (
            <Link className="btn" to="/auth">
              Вход / Регистрация
            </Link>
          )}
        </div>
      </div>
    </header>
  );
};

export default Header;
