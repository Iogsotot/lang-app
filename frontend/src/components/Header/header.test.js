import { BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Test suits for Header', () => {
  it('should match with snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
      <Header/>
      </BrowserRouter>
    )
    .toJSON();
   expect(tree).toMatchSnapshot();
   });
  });