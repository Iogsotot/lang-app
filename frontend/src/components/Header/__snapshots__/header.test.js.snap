// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Header should match with snapshot 1`] = `
<header>
  Header
</header>
`;
